### PR TITLE
Rendering: fix for case where StringNode contains quotes

### PR DESCRIPTION
**Pull request by yisraelU**

### CODE REVIEW

File: modules/codegen/src/smithy4s/codegen/Renderer.scala [hunk 1]
```diff
@@ -978,8 +978,10 @@ private[codegen] class Renderer(compilationUnit: CompilationUnit) { self =>
             }
             line"smithy4s.Document.obj(${members.toList.intercalate(Line.comma)})"
           }
-          def stringNode(x: StringNode): Line =
-            line"""smithy4s.Document.fromString("${x.getValue}")"""
+          def stringNode(x: StringNode): Line = {
+            val escapedString = x.getValue.replace("\"", "\\\"")
```
Comment (kubukoz) [inline modules/codegen/src/smithy4s/codegen/Renderer.scala:982]: I think this should use `renderStringLiteral`. It takes care of this and other cases (such as unicode characters, probably newlines and other escapes as well).

**Review by kubukoz — CHANGES_REQUESTED** [review]

File: modules/codegen/src/smithy4s/codegen/Renderer.scala (hunk 1 above)
Comment (yisraelU) [reply modules/codegen/src/smithy4s/codegen/Renderer.scala:982]: @kubukoz lol , you had me going crazy looking for that method.
thats on 0.17 branch,ill move it to main

Comment (kubukoz) [reply modules/codegen/src/smithy4s/codegen/Renderer.scala:982]: didn't notice this was targetting main :P 

are we actually going to keep using main though?

Comment (yisraelU) [reply modules/codegen/src/smithy4s/codegen/Renderer.scala:982]: dunno , all open pr's target main.
I assumed we would merge 0.17 into main

Comment (Baccata) [reply modules/codegen/src/smithy4s/codegen/Renderer.scala:982]: @yisraelU the new "default" branch is `series/0.17`, we'll be doing this from now on : the default branch will be tracking whatever current "major" version is (or "minor" version according to early-semver semantics)

Comment (yisraelU) [reply modules/codegen/src/smithy4s/codegen/Renderer.scala]: got it

**Comment by kubukoz** [comment]
I think  you're a couple commits behind, hence some extra changes. Please update and try to avoid unnecessary changes in unrelated files

**Comment by yisraelU** [comment]
@kubukoz something got screwed up for me locally , closing and opening new PR

**Pull request closed without merging**